src/pages/Registration.tsx

```diff
@@ -0,0 +1,117 @@
+import {ArrowBackIcon, EmailIcon, Icon} from '@chakra-ui/icons'
+import {
+    Button,
+    Center,
+    FormControl,
+    FormErrorMessage,
+    FormLabel,
+    Heading, HStack,
+    Input,
+    InputGroup,
+    InputRightAddon,
+    Text,
+    useToast,
+    VStack
+} from '@chakra-ui/react'
+import {
+    Field,
+    FieldProps,
+    Form,
+    Formik,
+    FormikProps,
+    FormikValues,
+} from 'formik'
+import React from 'react'
+import { useNavigate } from 'react-router-dom'
+import { useFetch } from 'use-http'
+import * as Yup from 'yup';
+import { object, string } from 'yup'
+
+export default function Registration() {
+    const {post, response} = useFetch()
+    const navigate = useNavigate()
+    const toast = useToast()
+    const [show, setShow] = React.useState(false)
+    const handleClick = () => setShow(!show)
+
+    const register = async (values: FormikValues) => {
+        await post('/register', values)
+        if (!response.ok && response.status == 409) {
+            toast({ title: 'Email already taken - are you already registered?', description: response.status, status: response.ok ? 'success' : 'error' })
+            return
+        }
+        navigate('/dashboard')
+    }
+
+
+    const schema = object({ email: string().required().matches(/^.*((?=.*[@]){1}).*$/, "Not a valid email"),
+        password: string().required().matches(
+            /^.*(?=.{6,})((?=.*[!@#$%^&*()\-_=+{};:,<.>?]){1})(?=.*\d)((?=.*[a-z]){1})((?=.*[A-Z]){1}).*$/,
```

Comment:
Yup has a built-in email validation so we can replace the regex pattern with string().email()